packages/beacon-node/src/api/impl/validator/utils.ts

```diff
@@ -44,31 +46,70 @@ export function getPubkeysForIndices(
   return pubkeys;
 }
 
-export function selectBlockProductionSource({
-  builderSelection,
-  engineBlockValue,
-  builderBlockValue,
-  builderBoostFactor,
-}: {
-  builderSelection: routes.validator.BuilderSelection;
-  engineBlockValue: bigint;
-  builderBlockValue: bigint;
-  builderBoostFactor: bigint;
-}): ProducedBlockSource {
+export function selectBlockProductionSource(
```

Comment:
should this function return a {source, reason} instead? pull metrics out?

Reply:
Good call